modules/fedimint-ecash-migration-client/src/lib.rs

```diff
@@ -57,28 +104,969 @@ impl ClientModule for EcashMigrationClientModule {
     ) -> Option<Amounts> {
-        unreachable!()
+        let fee_btc = match input {
+            fedimint_ecash_migration_common::EcashMigrationInput::RedeemOriginEcash { .. } => {
+                self.cfg.fee_config.transfer_redeem_fee
+            }
+            fedimint_ecash_migration_common::EcashMigrationInput::Default { .. } => {
+                unreachable!("We never produce default inputs")
+            }
+        };
+
+        Some(Amounts::new_bitcoin(fee_btc))
     }
 
     fn output_fee(
         &self,
         _amount: &Amounts,
-        _output: &<Self::Common as ModuleCommon>::Output,
+        output: &<Self::Common as ModuleCommon>::Output,
     ) -> Option<Amounts> {
-        unreachable!()
+        let fee_btc = match output {
+            EcashMigrationOutput::CreateTransfer(ecash_migration_create_transfer_output) => self
+                .cfg
+                .fee_config
+                .creation_fee(ecash_migration_create_transfer_output.spend_book_entries)
+                .expect("Can't overflow unless fee is unreasonable or spend book is ginormous"),
+            EcashMigrationOutput::FundTransfer(_) => self.cfg.fee_config.transfer_funding_fee,
+            EcashMigrationOutput::Default { .. } => {
+                unreachable!("We never produce default outputs")
+            }
+        };
+
+        Some(Amounts::new_bitcoin(fee_btc))
     }
 
     async fn get_balance(&self, _dbtx: &mut DatabaseTransaction<'_>, _unit: AmountUnit) -> Amount {
         Amount::ZERO
     }
+
+    #[cfg(feature = "cli")]
+    async fn handle_cli_command(
+        &self,
+        args: &[std::ffi::OsString],
+    ) -> anyhow::Result<serde_json::Value> {
+        cli::handle_cli_command(self, args).await
+    }
+}
+
+/// Metadata stored with the register liability transfer operation
+#[derive(Debug, Clone, Serialize, Deserialize, Encodable, Decodable)]
+pub struct RegisterTransferOperationMeta {
+    pub spend_book_merkle_root: MerkleRoot<Nonce>,
+    pub spend_book_entries: u64,
+    pub key_set_hash: KeySetHash,
+    pub out_point: OutPoint,
+}
+
+/// Metadata stored with the fund liability transfer operation
+#[derive(Debug, Clone, Serialize, Deserialize, Encodable, Decodable)]
+pub struct FundTransferOperationMeta {
+    pub transfer_id: TransferId,
+    pub amount: Amount,
+    pub change: OutPointRange,
+}
+
+/// Metadata stored with the redeem origin ecash operation
+#[derive(Debug, Clone, Serialize, Deserialize, Encodable, Decodable)]
+pub struct RedeemOriginEcashOperationMeta {
+    pub transfer_id: TransferId,
+    pub amount: Amount,
+    pub change: OutPointRange,
+}
+
+/// Read the origin federation's mint keyset from a config file.
+///
+/// The config file should be in JSON format as produced by
+/// `fedimint-cli --data-dir <data-dir> config` on the origin federation.
+///
+/// # Errors
+/// - If the origin config file cannot be read or parsed.
+pub async fn read_origin_keyset(
```

Comment:
Shouldn't this be server side command? Because later down the line we might do migrations and change stuff and then matching client version to server version will be a pain, while it's easy to tell guardian to use the same binary they were using for running the federation in the first place, no? IIRC joshisan wanted to eventually move configs into db, or the other way around - point being that potentially these are moving targets.

We could have some `fedimintd export-post-mortem --data-dir ...` that would create some versioned (so we can change the format) "export file", and possibly an guardian-authed api endpoint that can do the same thing, or something, and in the client we would be reading the standardized format only, not snoop in the server internals.